gammapy/data/hdu_index_table.py

```diff
@@ -222,20 +222,20 @@ def row_idx(self, obs_id, hdu_type=None, hdu_class=None):
         idx : list of int
             List of row indices matching the selection.
         """
-        # Working with the HDU_CLASS or HDU_TYPE column directly is difficult,
-        # because those are padded strings (sometimes left-padded, sometimes right-padded).
-        # Using a Python loop and cached lists of stripped string columns is easiest here.
-        idx_list = []
+        selection = self['OBS_ID'] == obs_id
 
-        for idx in range(len(self)):
-            if self['OBS_ID'][idx] == obs_id:
-                if hdu_class and self._hdu_class_stripped[idx] == hdu_class:
-                    idx_list.append(idx)
+        if hdu_class:
+            hdu_class_stripped = np.array(self._hdu_class_stripped)
```

Comment:
If you stick with this implementation: why have the Python list -> numpy array conversion in the lookup?
Make the cached property a numpy array?